articles/azure-app-configuration/enable-dynamic-configuration-dotnet-core-push-refresh.md

```diff
@@ -175,6 +175,112 @@ A random delay is added before the cached value is marked as dirty to reduce pot
 > [!NOTE]
 > To reduce the number of requests to App Configuration when using push refresh, it is important to call `SetCacheExpiration(TimeSpan cacheExpiration)` with an appropriate value of `cacheExpiration` parameter. This controls the cache expiration time for pull refresh and can be used as a safety net in case there is an issue with the Event subscription or the Service Bus subscription. The recommended value is `TimeSpan.FromDays(30)`.
 
+## Push Model
+
+> [!NOTE]
+> The API's used in this example are only available in the provider library version 5.x and later.
+
+This is an optional section which works with the Push Model. Using a `SyncToken` ensures that users will receive their most up to date configurations. This `SyncToken` is sent to the `Azure App Configuration` SDK which is subsequently passed to `Cosmos DB` to ensure the most recent configurations are returned. This new model prevents replication lag from occurring.
```

Comment:
We can simplify this section without describing too many implementation details like CosmosDB, or any of the new APIs we added (those are already documented in our API reference docs). Suggested change:

Due to the nature of distributed systems, real-time consistency between requests is difficult to enforce implicitly. We use synchronization tokens to address this issue. The Event Grid event contains the sync-token that can be utilized to guarantee real-time consistency between different client instances and requests. 

To use sync-token in your push refresh workflow, you need to make the following changes to your  **Program.cs** file:

1. Add a dependency on Azure.Messaging.EventGrid Package (add code snippet). 
2. Make changes to the RegisterRefreshEventHandler method (add code snippet).

(After code snippets, add one last paragraph about how ProcessPushNotification calls SetDirty internally)